replay/models/base_rec.py

```diff
@@ -457,33 +456,6 @@ def _filter_seen(
         """
         users_log = log.join(users, on="user_idx")
         self._cache_model_temp_view(users_log, "filter_seen_users_log")
-        num_seen = users_log.groupBy("user_idx").agg(
```

Comment:
t seems those steps are not redundant.
some models (a lot of models) do not filter seen items inside _predict(). So if we want to predict with filter_seen_items=True, we

predict more items than needed (k + max_seen),
filter out seed items (thus we will get number of recommendations in interval [k, k + max_seen]
get top-k most relevant
So, ether you placed seen filtering inside _predict method of each model and it is not in the current pr, or you just removed some necessary functionality.